File: src/main/java/com/ninebudget/component/InstitutionComponent.java
package com.ninebudget.component;

import com.ninebudget.model.Component;
import com.ninebudget.model.ComponentException;
import com.ninebudget.model.Institution;

public class InstitutionComponent implements Component<Institution> {
    @Override
    public Institution get(Institution object) throws ComponentException {
        return null;
    }

    @Override
    public void update(Institution object) throws ComponentException {

    }

    @Override
    public void create(Institution object) throws ComponentException {

    }

    @Override
    public void deactivate(Institution object) throws ComponentException {

    }
}

File: web/src/app/component/institution/institution.component.ts
import {Component, OnInit} from '@angular/core';
import {Institution} from "../../model/institution";
import {InstitutionService} from "../../service/institution.service";
import {MessageService} from "primeng/api";

;

@Component({
  selector: 'app-institution',
  templateUrl: './institution.component.html',
  styleUrls: ['./institution.component.scss']
})
export class InstitutionComponent implements OnInit {
  institutions?: Institution[];

  constructor(private institutionService: InstitutionService
    , private messageService: MessageService) {
  }

  ngOnInit() {

    this.institutionService.getAll().subscribe(
      data => {
        this.institutions = data ? data : [];
        this.messageService.add({summary: 'Institution Loaded', severity: 'success', sticky: false})
      },
      error => this.messageService.add({summary: 'Failed to Institution: ' + error, severity: 'error', sticky: false})
    );


  }

}
